cedar-lean/DiffTest/Parser.lean

```diff
@@ -165,57 +169,68 @@ partial def jsonToExpr (json : Lean.Json) : ParseResult Expr := do
     .ok (.or lhs rhs)
   | "If" => do
-    let i ← getJsonField body "test_expr" >>= jsonToExpr
-    let t ← getJsonField body "then_expr" >>= jsonToExpr
-    let e ← getJsonField body "else_expr" >>= jsonToExpr
+    let i ← getJsonField body "test_expr" >>= jsonToPartialExpr
+    let t ← getJsonField body "then_expr" >>= jsonToPartialExpr
+    let e ← getJsonField body "else_expr" >>= jsonToPartialExpr
     .ok (.ite i t e)
   | "UnaryApp" => do
     let op ← getJsonField body "op" >>= jsonToUnaryOp
-    let arg ← getJsonField body "arg" >>= jsonToExpr
+    let arg ← getJsonField body "arg" >>= jsonToPartialExpr
     .ok (.unaryApp op arg)
   | "Like" => do
     let pat ← getJsonField body "pattern" >>= jsonToPattern
-    let expr ← getJsonField body "expr" >>= jsonToExpr
+    let expr ← getJsonField body "expr" >>= jsonToPartialExpr
     .ok (.unaryApp (.like pat) expr)
   | "Is" => do
     let ety ← getJsonField body "entity_type" >>= jsonToName
-    let expr ← getJsonField body "expr" >>= jsonToExpr
+    let expr ← getJsonField body "expr" >>= jsonToPartialExpr
     .ok (.unaryApp (.is ety) expr)
   | "BinaryApp" => do
     let op ← getJsonField body "op" >>= jsonToBinaryOp
-    let arg1 ← getJsonField body "arg1" >>= jsonToExpr
-    let arg2 ← getJsonField body "arg2" >>= jsonToExpr
+    let arg1 ← getJsonField body "arg1" >>= jsonToPartialExpr
+    let arg2 ← getJsonField body "arg2" >>= jsonToPartialExpr
     .ok (.binaryApp op arg1 arg2)
   | "GetAttr" => do
-    let e ← getJsonField body "expr" >>= jsonToExpr
+    let e ← getJsonField body "expr" >>= jsonToPartialExpr
     let attr ← getJsonField body "attr" >>= jsonToString
     .ok (.getAttr e attr)
   | "HasAttr" => do
-    let e ← getJsonField body "expr" >>= jsonToExpr
+    let e ← getJsonField body "expr" >>= jsonToPartialExpr
     let attr ← getJsonField body "attr" >>= jsonToString
     .ok (.hasAttr e attr)
   | "Record" => do
     let kvs_json ← jsonObjToKVList body
-    let kvs ← mapMValues kvs_json jsonToExpr
+    let kvs ← mapMValues kvs_json jsonToPartialExpr
     .ok (.record kvs)
   | "Set" => do
     let arr_json ← jsonToArray body
-    let arr ← List.mapM jsonToExpr arr_json.toList
+    let arr ← List.mapM jsonToPartialExpr arr_json.toList
     .ok (.set arr)
   | "ExtensionFunctionApp" => do
     let fn ← getJsonField body "fn_name" >>= jsonToExtFun
     let args_json ← getJsonField body "args" >>= jsonToArray
-    let args ← List.mapM jsonToExpr args_json.toList
+    let args ← List.mapM jsonToPartialExpr args_json.toList
     .ok (.call fn args)
-  | tag => .error s!"jsonToExpr: unknown tag {tag}"
+  | "Unknown" => do
+    let name ← getJsonField body "name" >>= jsonToString
+    .ok (.unknown name)
+  | tag => .error s!"jsonToPartialExpr: unknown tag {tag}"
+
+def jsonToExpr (json : Lean.Json) : ParseResult Expr := do
+  let pe ← jsonToPartialExpr json
+  match pe.asSpec with
+  | some e => .ok e
+  | none => .error s!"expression contained unknown"
```

Comment:
```suggestion
  | none => .error s!"jsonToExpr: expression contained unknown"
```
When debugging in the past, I've found it useful to include the function name in the error message.